Prox/Prox/Data/PlacesController.swift

```diff
@@ -257,14 +270,20 @@ extension PlacesProvider: PlaceDataSource {
         }
     }
 
-    func allPlaces() -> [Place] {
-        return displayedPlaces
-    }
-
     func sortPlaces(byLocation location: CLLocation) {
         self.placesLock.withWriteLock {
             let sortedPlaces = PlaceUtilities.sort(places: displayedPlaces, byDistanceFromLocation: location)
             self.displayedPlaces = sortedPlaces
         }
     }
+
+    func refresh() {
+        assert(Thread.isMainThread)
+
+        placesLock.withWriteLock {
+            updateDisplayedPlaces()
+        }
+
+        delegate?.placesProvider(self, didUpdatePlaces: displayedPlaces)
```

Comment:
We take a read lock when we call this in `displayPlaces`. As I mention earlier, maybe we should just return a copy.